core/cloudflow-cr-generator/src/main/scala/cloudflow/cr/Generator.scala

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (C) 2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package cloudflow.cr
+
+import com.typesafe.config.{ ConfigFactory, ConfigRenderOptions }
+
+import java.io.File
+import java.net.URL
+import scala.collection.JavaConverters._
+import scala.io.Source
+import scala.util.{ Failure, Success, Try }
+
+import scopt.OParser
+import spray.json._
+import cloudflow.blueprint._
+import cloudflow.blueprint.StreamletDescriptorFormat._
+import cloudflow.blueprint.deployment._
+import cloudflow.extractor._
+
+object Generator {
+
+  val cloudflowVersion = BuildInfo.version
+
+  private def makeCR(appDescriptor: ApplicationDescriptor, version: String): CloudflowCR =
+    CloudflowCR(
+      apiVersion = "cloudflow.lightbend.com/v1alpha1",
+      kind = "CloudflowApplication",
+      metadata = Metadata(
+        annotations = Map("com.lightbend.cloudflow/created-by-cli-version" -> version),
+        labels = Map(
+          "app.kubernetes.io/managed-by" -> "cloudflow",
+          "app.kubernetes.io/part-of" -> appDescriptor.appId,
+          "com.lightbend.cloudflow/app-id" -> appDescriptor.appId),
+        name = appDescriptor.appId),
+      appDescriptor)
+
+  def scanProject(projectId: String, classpath: List[String]): Map[String, com.typesafe.config.Config] = {
+    val scanConfig =
+      DescriptorExtractor.ScanConfiguration(projectId = projectId, classpathUrls = classpath.map(new URL(_)).toArray)
+
+    DescriptorExtractor.scan(scanConfig)
+  }
+
+  def generate(
+      projectId: String,
+      version: String,
+      blueprintStr: String,
+      streamlets: Map[String, com.typesafe.config.Config],
+      dockerImages: Map[String, String]): String = {
```

Comment:
May be return a `Try[String]` instead of throwing exceptions ?

Reply:
The error messages are quite nice from Maven in case we directly throw exceptions!

Reply:
But what happens if downstream fails to catch those nice exceptions ? :-)

Reply:
> downstream fails to catch those nice exceptions

I'm expecting downstream to NOT catch those exceptions and to be propagated up to the JVM (which craches), this is the contract, isn't it?